state/migration_export.go

```diff
@@ -482,15 +483,25 @@ func (e *exporter) applications() error {
 		return errors.Trace(err)
 	}
 
+	resourcesSt, err := e.st.Resources()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
 	for _, application := range applications {
 		applicationUnits := e.units[application.Name()]
 		leader := leaders[application.Name()]
+		resources, err := resourcesSt.ListResources(application.Name())
```

Comment:
I *might* be able to avoid the extra lookups but it'll mean extending or going around the many levels of resources state API.